### DISCUSSION TITLE
Organization of the material parameters

### DISCUSSION THREAD

**Discussion by HSDSZ** [bnpr/Malt#187] [Feedback & Feature Requests]
Any way to have an organized parameters while keeping those ticks 

If I use node. It is good that I can have a tick on the right to overwrite. But is is not good that the useless information such as the  "v" and "f" are still here. 
![image](https://user-images.githubusercontent.com/36077492/143926253-590e50a8-ff0d-41f6-b134-deeea4905dc4.png)

If I use a custom struct, Parameters are organized and the tick overwrite is still there . But I have two "Outline" title, which looks strange
![image](https://user-images.githubusercontent.com/36077492/143926014-7354a85f-61e4-45ab-a148-d5ca598a7c2e.png)

If I use code for shader, the parameters are well organized, But I no longer have that tick button for overwrite
![image](https://user-images.githubusercontent.com/36077492/143927166-698238e7-2210-4b1a-aa61-3c56a651775c.png)

How to achieve the "code" group style  via node.

**Comment by pragma37** [comment]
I think in the second case it should be possible to avoid the duplication.
I'll look into it.

**Reply by HSDSZ** [reply]
Actually, supporting custom variable names in "inline code" is enough for most cases.
